src/hex.py

```diff
@@ -13,10 +13,18 @@
 match command:
     case "encode":
         # Implement the encoding here
-        encoding = ""
+        y = []
+        for letter in x: 
+            y.append(hex(ord(letter)))
+        encoding = ''.join(y)
         print(encoding)
 
     case "decode":
         # Implement the decoding here
-        decoding = ""
+        for letter in x:
+            new = x.split('0x')
+            string = ''.join(new[1:])
+            integer = int(string, base = 16)
+        decoding = chr(integer)
```

Comment:
I find it hard to follow the reasoning here. If you don't change `x`, then `new` gets the same value in each iteration. That means `string` gets the same value, so `integer` does as well.